### PR TITLE
fix(snql): Add aggregations to select in auto_aggregation

### PR DESCRIPTION
**Pull request by wmak**
- This is to fix an issue for queries that have the uniq aggregation in
  the HAVING clause, and is not selected.
  - Previously we would not add the aggregation to the select clause in
    these cases
  - Now anything in the having clause will get added to the select
    clause as well if auto_aggregation is enabled
  - if its disabled we raise an invalid search query error
- This also fixes a bug where this having validation wasn't working
  correctly for boolean conditions

### CODE REVIEW

File: src/sentry/search/events/builder.py [hunk 1]
```diff
@@ -114,18 +114,43 @@ def validate_having_clause(self) -> None:
...
+            elif isinstance(condition, BooleanCondition):
+                boolean_conditions.append(condition)
+
+        while len(boolean_conditions) > 0:
+            boolean_condition = boolean_conditions.pop()
+            for condition in boolean_condition.conditions:
+                if isinstance(condition, Condition):
+                    to_check.append(condition)
+                elif isinstance(condition, BooleanCondition):
+                    boolean_conditions.append(condition)
```
Comment (Zylphrex) [inline src/sentry/search/events/builder.py:133]: Nit: refactor this into a helper function

File: src/sentry/search/events/builder.py [hunk 2]
```diff
@@ -114,18 +114,43 @@ def validate_having_clause(self) -> None:
...
+
         if self.auto_aggregations and self.aggregates:
+            for condition in to_check:
+                lhs = condition.lhs
+                if isinstance(lhs, CurriedFunction) and lhs not in self.columns:
+                    self.columns.append(lhs)
+                    self.aggregates.append(lhs)
```
Comment (Zylphrex) [inline src/sentry/search/events/builder.py:140]: Why do `lhs` need to be added to `self.aggregates`?

**Review by Zylphrex — COMMENTED** [review]
Do we need to do this validation in all the other builders like `TimeseriesBuilder`?

File: src/sentry/search/events/builder.py (hunk 2 above)
Comment (wmak) [reply src/sentry/search/events/builder.py:140]: Not used anywhere, added this for consistency since `lhs` is an aggregation

**Comment by wmak** [comment]
> Do we need to do this validation in all the other builders like TimeseriesBuilder?

No for all of the above, but going through each one just so you can check my math:
- `TimeseriesBuilder` -> `use_aggregate_conditions=False` which means `self.having` will always be empty
- `TopEventsQueryBuilder` -> Gets the top-n by calling the base QueryBuilder, but the conditions we construct are fields only and don't use the aggregates
- `HistogramQueryBuilder` -> Inherits from QueryBuilder, so no extra work needed

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**